File: Dockerfile.java
FROM openjdk:11

# Bad: No multi-stage build, remains in a large JDK image
WORKDIR /app

# Bad: COPY all source code
COPY . .

# Bad: Running build inside the final image
RUN ./gradlew build

# Bad: Running with standard jar (heavy environment)
EXPOSE 8080
CMD ["java", "-jar", "build/libs/app.jar"]

File: Dockerfile.go
FROM golang:1.18

# Bad: No multi-stage build, keeping the entire Go toolchain
WORKDIR /go/src/app
COPY . .

# Bad: Fetching dependencies in a separate layer without caching optimization
RUN go get -d -v ./...
RUN go install -v ./...

# Bad: Final image is unnecessarily large
CMD ["app"]
